airflow/task/task_runner/base_task_runner.py

```diff
@@ -161,4 +161,4 @@ def on_finish(self):
         A callback that should be called when this is done running.
         """
         if self._cfg_path and os.path.isfile(self._cfg_path):
-            subprocess.call(['sudo', 'rm', self._cfg_path], close_fds=True)
+            subprocess.call(['rm', self._cfg_path], close_fds=True)
```

Comment:
could the case happen when the dir is created and deleted by different users?

Reply:
so the temp cfg file is owned by the same user who runs airflow worker command, for example:
```
jundayang@airflowinfraworker-production-iad-0184b3e28b386f0cf:/tmp$ ls -la tmpzyfuiw2v

-rw------- 1 airflow airflow 7139 Mar 13 21:25 tmpzyfuiw2v
```
At least we need to allow user who creates the file to delete the file?

Reply:
the command failed was not due to `rm`, but actually due to `sudo` based on the error:
```
Mar 13 21:22:33 : airflow : user NOT in sudoers ; TTY=unknown ; PWD=/srv/airflowinfra ; USER=root ; COMMAND=/bin/rm /tmp/tmpgzw8eaww
```

Reply:
![menu saveimg savepath20190314164828](https://user-images.githubusercontent.com/28775990/54342973-b0ef3f80-4678-11e9-8df3-61b7b96269b0.jpg)

Excuse me  I've got a problem. I'm going to execute once , but it was executed multiple times
 
cat  airflow.cfg
executor = LocalExecutor

Reply:
We still need this when `self.run_as_user is Not None`
```suggestion
            if self.run_as_user:
                subprocess.call(['rm', self._cfg_path], close_fds=True)
            else:
                os.remove(self._cfg_path)
```

Reply:
@ashb  I assume what you mean is:
```
if self.run_as_user:
--
  | subprocess.call(['sudo','rm', self._cfg_path], close_fds=True)
  | else:
  | os.remove(self._cfg_path)
```

Reply:
@ashb this makes sense to me after knowing more about `run_as_user`. You mean if the `airflow worker` command is run by `root` and the `run_as_user` at the same time needs to have sudo privilege to call `sudo rm /tmp/...` to remove the temp cfg file owned by root, right?

btw, I think you missed `sudo` in `subprocess.call(['rm', self._cfg_path], close_fds=True)` of your suggestion, right?

Reply:
Or even if airflow runs as `airflow` but a task runs as `hadoop` unix user it'll be needed.